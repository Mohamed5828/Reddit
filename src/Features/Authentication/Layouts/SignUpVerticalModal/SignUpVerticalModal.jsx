import { useState } from "react";

import Modal from "react-bootstrap/Modal";

import {
  MyModal,
  MyModalHeader,
  HaveAccount,
} from "./SignUpVerticalModal.styled";

import SignUp from "Features/Authentication/Layouts/SignUp/SignUp";
import LogIn from "Features/Authentication/Layouts/LogIn/LogIn";

/**
 * SignUpVerticalModal component that is used to show the modal of signup
 * @returns {React.Component}  SignUpVerticalModal component that is used to show the modal of signup
 */

const SignUpVerticalModal = ({ ...props }) => {
  /**
   * state to know what component should be shown in the modal
   */
  const [showLogIn, setShowLogIn] = useState(false);

  return (
    <MyModal
      id="signUpModal"
      {...props}
      size="lg"
      aria-labelledby="contained-modal-title-vcenter"
      centered
      onExit={() => {
        setShowLogIn(false);
      }}
    >
      <MyModalHeader closeButton></MyModalHeader>
      <Modal.Body>
        {!showLogIn && (
          <>
            <SignUp />{" "}
            <HaveAccount>
              Already a redditor?{" "}
              <button
                id="logInButtonFromSignupModal"
                onClick={() => setShowLogIn(true)}
              >
                Log in
              </button>
            </HaveAccount>
          </>
        )}
        {showLogIn && (
          <>
            {" "}
            <LogIn />{" "}
            <HaveAccount>
              New to Reddit?{" "}
              <button onClick={() => setShowLogIn(false)}>Sign up</button>
            </HaveAccount>
          </>
        )}
      </Modal.Body>
    </MyModal>
  );
};

export default SignUpVerticalModal;
