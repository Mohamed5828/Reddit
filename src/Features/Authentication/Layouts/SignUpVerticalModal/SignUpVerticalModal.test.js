import { shallow } from "enzyme";

import SignUpVerticalModal from "./SignUpVerticalModal";




describe("SignUpVerticalModal Modal", () => {
  it("this is a test for SignUpVerticalModal Modal", () => {
    expect(shallow(<SignUpVerticalModal />)).toMatchSnapshot();
  });
});
