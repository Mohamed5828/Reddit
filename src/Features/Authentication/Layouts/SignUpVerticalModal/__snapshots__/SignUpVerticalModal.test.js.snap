// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpVerticalModal Modal this is a test for SignUpVerticalModal Modal 1`] = `
<Styled(Modal)
  aria-labelledby="contained-modal-title-vcenter"
  centered={true}
  id="signUpModal"
  onExit={[Function]}
  size="lg"
>
  <Styled(ModalHeader)
    closeButton={true}
  />
  <ModalBody>
    <SignUp />
     
    <styled.span>
      Already a redditor?
       
      <button
        id="logInButtonFromSignupModal"
        onClick={[Function]}
      >
        Log in
      </button>
    </styled.span>
  </ModalBody>
</Styled(Modal)>
`;
